### PR TITLE
PROXY-6

### PR DESCRIPTION
**Pull request by xiangzhe0812**
1.setup db and docker
2.add end point to push expo notification token.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: "2.2"
+services:
+    postgres:
+        image: postgres
+        environment:
+            POSTGRES_DB: idem
+            POSTGRES_USER: idem
+            POSTGRES_PASSWORD: test123
```
Comment (bitcoinbrisbane) [inline docker-compose.yml:8]:
```suggestion
            POSTGRES_PASSWORD: Test1234
```

File: src/data/migrations/CNew-01-project-setup.sql [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+CREATE OR REPLACE FUNCTION update_updatedAt()
+RETURNS TRIGGER AS $$
+BEGIN
+   NEW."updatedAt" = now();
+   RETURN NEW;
+END;
+$$ language "plpgsql";
+
+CREATE TABLE "users" (
```
Comment (xcjlo985) [inline src/data/migrations/CNew-01-project-setup.sql:11]: Make this idempotent

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+version: "2.2"
```
Comment (xcjlo985) [inline docker-compose.yml:1]: * Update the readme with steps on how/when/why to run this
* Can you make sure it's the version of PG that's the same as the managed DB?

File: src/data/database.providers.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import { DataSource } from "typeorm";
+
+export const databaseProviders = [
+    {
+        provide: "DATA_SOURCE",
+        useFactory: async () => {
+            const dataSource = new DataSource({
```
Comment (xcjlo985) [inline src/data/database.providers.ts:7]: Make this a pooled datasource?

File: src/user/user.service.ts [hunk 5]
```diff
@@ -1,4 +1,35 @@
-import { Injectable } from "@nestjs/common";
+import { Injectable, Inject } from "@nestjs/common";
+import { User } from "../data/entities/user.entity";
```
Comment (xcjlo985) [inline src/user/user.service.ts:2]: Nitpick: unrelated but an improvement to remove relative paths (aka the devil): https://medium.com/@fmoessle/typescript-paths-with-ts-node-ts-node-dev-and-jest-671deacf6428

**Review by xcjlo985 — APPROVED** [review]

**Review by jrocca82 — APPROVED** [review]

**Pull request merged**